src/Chapter3.hs

```diff
@@ -970,6 +1058,21 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay = Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday deriving (Show,Read,Eq,Ord,Enum)
+
+isWeekend::WeekDay->Bool
+isWeekend d 
+  | d == Sunday || d == Saturday = True
+  | otherwise = False
+
+nextDay :: WeekDay->WeekDay
+nextDay Saturday = Sunday
+nextDay d = toEnum $ fromEnum d + 1
+
+daysToParty:: WeekDay->Int
+daysToParty Friday = 0
+daysToParty d = 1 + daysToParty ( nextDay d) -- (daysToParty Saturday) == 6 -- but that's the spec
```

Comment:
That's nice recursive solution! But you can implement it a bit more efficiently and save a few CPU cycles by using the `fromEnum` method and `mod`. Remember how `mod` works for negative numbers from Chapter One?